openpathsampling/beta/hooks.py

```diff
@@ -78,12 +143,42 @@ class ShootFromSnapshotsOutputHook(PathSimulatorHook):
     def __init__(self, output_stream=None, allow_refresh=None):
         self.output_stream = output_stream
         self.allow_refresh = allow_refresh
+        self._simulation = None
+
+    @property
+    def output_stream(self):
+        if self._output_stream is not None:
+            return self._output_stream
+        elif self._simulation is not None:
+            return self._simulation.output_stream
+        else:
+            raise SimulationNotFoundError("'output_stream' has not "
```

Comment:
(Starting to glance over it, but a thought before you spend much time on remaining tests.) You might to create something like

```python
def _self_or_sim_property(obj, attr_name):
    if getattr(obj, "_" + attr_name) is not None:
        return getattr(obj, "_" + attr_name)
    ...  # etc.
```

then use it as

```python
    @property
    def output_stream(self):
        return _self_or_sim_property(self, 'output_stream')
```

This would allow you to test the underlying logic once, and not need to worry about creating multiple tests for each property. More complete integration test are also welcome, but this definitely seems like a motif that is being (and will continue to be) repeated.

Reply:
(You could add a `return_none=False` to that private method to make it return `None` instead of erroring when that it appropriate)